### ISSUE TITLE
Bug: $convertFromMarkdownString method from @lexical/markdown ignores linebreaks

### ISSUE THREAD

**Issue by lbarrous** [facebook/lexical#6138]
Issue edited:

`$convertFromMarkdownString` is avoiding multiple "\n", in this case for example:

If I provide as an initial content something like this:

`editorState: () =>
      $convertFromMarkdownString(
        'ab\n\nc\n\n\n\n - item\n- item2\n1. item1\n2. item2\n3. item3\n\nTest\n\nTest\n- another list\n- with items\n-  {{additional_information}} ',
        TRANSFORMERS,
      ),`
     
I expect to have some output like this:
![image](https://github.com/facebook/lexical/assets/12309518/a90ead61-19ce-4407-8751-0bb098387032)

However I get this:
![image](https://github.com/facebook/lexical/assets/12309518/1632ab7a-c0a3-452c-855b-b6fc3e820abf)

Lexical version: Latest

## The expected behavior

Output is properly rendered and it must be as the first screenshot provided.

**Comment by 2wheeh** [comment]
> Which basically, after being converted to markdown, it is something like this:
>
> `- item\n- item2\n1. item1\n2. item2\n3. item3\nTest\nTest\n- another list\n- with items\n`

I think your input is converted to like this:
`- item\n- item2\n\n1. item1\n2. item2\n3. item3\n\nTest\n\nTest\n\n- another list\n- with items\n`

**Comment by etrepum** [comment]
This is how GitHub renders the markdown that you've provided, which is consistent with the screenshot in this report:

---

- item
- item2
1. item1
2. item2
3. item3
Test
Test
- another list
- with items

---

I made sure to reproduce it exactly by using this from the browser console, and then pasting it into this comment:

```js
copy('- item\n- item2\n1. item1\n2. item2\n3. item3\nTest\nTest\n- another list\n- with items\n')
```

**Comment by lbarrous** [comment]
Yes, you're right, I was deeping into my code and I saw somehow I was using in the middle of the process a format function which was basically replacing "\n\n" by "\n", which was causing this. 

Now, anyway I figured out about another issue, which is basically that `$convertFromMarkdownString` is avoiding multiple "\n", in this case for example:

If I provide as an initial content something like this:

`editorState: () =>
      $convertFromMarkdownString(
        'ab\n\nc\n\n\n\n - item\n- item2\n1. item1\n2. item2\n3. item3\n\nTest\n\nTest\n- another list\n- with items\n-  {{additional_information}} ',
        TRANSFORMERS,
      ),`
     
I expect to have some output like this:
![image](https://github.com/facebook/lexical/assets/12309518/a90ead61-19ce-4407-8751-0bb098387032)

However I get this:
![image](https://github.com/facebook/lexical/assets/12309518/1632ab7a-c0a3-452c-855b-b6fc3e820abf)

Does it mean that line breaks somehow are ignored?

@2wheeh @etrepum 

Thanks!

**Comment by 2wheeh** [comment]
There would be new feature coming about `preserveNewLines` (#6020)
It seems not released yet though it's live on playground.

**Comment by lbarrous** [comment]
I saw on documentation https://lexical.dev/docs/api/modules/lexical_markdown that there's a `shouldPreserveNewLines` property in the signature.
![image](https://github.com/facebook/lexical/assets/12309518/d6747456-d9ae-4082-b734-50fa7912ec1e)

However in my current export of the method, it only has 3 arguments:
![image](https://github.com/facebook/lexical/assets/12309518/a116af07-0052-4d9d-88a5-0b1146ae81a2)

**Comment by lbarrous** [comment]
> There would be new feature coming about `preserveNewLines` (#6020) It seems not released yet though it's live on playground.
> 
> <img alt="image" width="201" src="https://private-user-images.githubusercontent.com/40269597/332819137-6213dfdf-a13e-4b9c-a903-510e3d3533b2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYzODU0NDIsIm5iZiI6MTcxNjM4NTE0MiwicGF0aCI6Ii80MDI2OTU5Ny8zMzI4MTkxMzctNjIxM2RmZGYtYTEzZS00YjljLWE5MDMtNTEwZTNkMzUzM2IyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIyVDEzMzkwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJhY2UzOTgzNWM2ZTk0YTY4MDdjMzU0Y2NhOTQ2NDcwYTBmNTk0OTBhOWZmODZlZDYxNjVhNjBiZGFhYTlmMTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NgrwsoMIhqNEodPvZyneVhYzkx8OOXT-f6jJj3dsmSY">

Oh,  thanks mate, I see, I guess I just need for that new feature to be released on a new version right?

**Comment by lbarrous** [comment]
is there a way to know when a new version is released to check for this feature?

**Comment by 2wheeh** [comment]
well, giving a star to this repo starts updating release news on your github feed.
there is also [discord community](https://discord.gg/pmMrTJprmW).

**Comment by illing2005** [comment]
Do you know the timeline for releasing this?

**Comment by potatowagon** [comment]
yes that correct, `shouldPreserveNewLines` flag would resolve this issue, documentation in 

https://lexical.dev/docs/api/modules/lexical_markdown#converttomarkdownstring

it is now available in [v0.16.0](https://github.com/facebook/lexical/releases/tag/v0.16.0)
